the trait bound `FieldAttr: ToTokens` is not satisfied (Installing the CLI)

<!--
If you want to report a bug, we added some points below which help us track down the problem faster.
-->

## Setup

### Versions

- **Rust:** 1.68.0 (2c8cc3432 2023-03-06)
- **Diesel:** 2.0.0
- **Database:** Postgres 
- **Operating System** MacOS arm64

### Feature Flags

- **diesel:**

## Problem Description

I get an error while trying to install the CLI (as described on the setup guide at [this link](https://diesel.rs/guides/getting-started.html)). 

### What are you trying to accomplish?

To setup diesel, to install the CLI tool.

### What is the expected output?

A success message?

### What is the actual output?

```
cargo install diesel_cli --no-default-features --features postgres
    Updating crates.io index
  Installing diesel_cli v2.0.1
   Compiling proc-macro2 v1.0.52
   Compiling unicode-ident v1.0.8
   Compiling quote v1.0.24
   Compiling syn v1.0.109
   Compiling autocfg v1.1.0
   Compiling serde_derive v1.0.155
   Compiling serde v1.0.155
   Compiling version_check v0.9.4
   Compiling memchr v2.5.0
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
   Compiling libc v0.2.140
   Compiling num-traits v0.2.15
   Compiling indexmap v1.9.2
   Compiling tinyvec_macros v0.1.1
   Compiling pq-sys v0.4.7
   Compiling bitflags v1.3.2
   Compiling core-foundation-sys v0.8.3
   Compiling tinyvec v1.6.0
   Compiling num-integer v0.1.45
   Compiling hashbrown v0.12.3
   Compiling os_str_bytes v6.4.1
   Compiling clap_lex v0.2.4
   Compiling atty v0.2.14
   Compiling aho-corasick v0.7.20
   Compiling unicode-bidi v0.3.11
   Compiling unicode-normalization v0.1.22
   Compiling termcolor v1.2.0
   Compiling regex-syntax v0.6.28
   Compiling itoa v1.0.6
   Compiling byteorder v1.4.3
   Compiling textwrap v0.16.0
   Compiling overload v0.1.1
   Compiling percent-encoding v2.2.0
   Compiling strsim v0.10.0
   Compiling nu-ansi-term v0.46.0
   Compiling clap v3.2.23
   Compiling form_urlencoded v1.1.0
   Compiling idna v0.3.0
   Compiling iana-time-zone v0.1.53
   Compiling diffy v0.3.0
   Compiling regex v1.7.1
   Compiling chrono v0.4.24
   Compiling clap_complete v3.2.5
   Compiling url v2.3.1
   Compiling diesel_derives v2.0.1
error[E0277]: the trait bound `FieldAttr: ToTokens` is not satisfied
   --> /Users/harry/.cargo/registry/src/github.com-1ecc6299db9ec823/diesel_derives-2.0.1/src/attrs.rs:143:18
    |
143 | impl Spanned for FieldAttr {
    |                  ^^^^^^^^^ the trait `ToTokens` is not implemented for `FieldAttr`
    |
    = help: the following other types implement trait `ToTokens`:
              &'a T
              &'a mut T
              Abi
              Abstract
              AndAnd
              AndEq
              AngleBracketedGenericArguments
              Arm
            and 312 others
    = note: required for `FieldAttr` to implement `quote::spanned::private::Sealed`
note: required by a bound in `quote::spanned::Spanned`
   --> /Users/harry/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.24/src/spanned.rs:6:20
    |
6   | pub trait Spanned: private::Sealed {
    |                    ^^^^^^^^^^^^^^^ required by this bound in `Spanned`

error[E0277]: the trait bound `StructAttr: ToTokens` is not satisfied
   --> /Users/harry/.cargo/registry/src/github.com-1ecc6299db9ec823/diesel_derives-2.0.1/src/attrs.rs:242:18
    |
242 | impl Spanned for StructAttr {
    |                  ^^^^^^^^^^ the trait `ToTokens` is not implemented for `StructAttr`
    |
    = help: the following other types implement trait `ToTokens`:
              &'a T
              &'a mut T
              Abi
              Abstract
              AndAnd
              AndEq
              AngleBracketedGenericArguments
              Arm
            and 312 others
    = note: required for `StructAttr` to implement `quote::spanned::private::Sealed`
note: required by a bound in `quote::spanned::Spanned`
   --> /Users/harry/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.24/src/spanned.rs:6:20
    |
6   | pub trait Spanned: private::Sealed {
    |                    ^^^^^^^^^^^^^^^ required by this bound in `Spanned`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `diesel_derives` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `diesel_cli v2.0.1`, intermediate artifacts can be found at `/var/folders/wx/y49q6sc950v5mhfvqkrc09100000gn/T/cargo-installYNdmN4`
```

### Are you seeing any additional errors?

No

### Steps to reproduce

The steps that are on the website. 

## Checklist

- [X] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues)  and the [disussion forum](https://github.com/diesel-rs/diesel/discussions)  for similar possible closed issues.
<!--
If you are unsure if your issue is a duplicate of an existing issue please link the issue in question here
--> 
- [X] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)
- [X] This issue can be reproduced without requiring a third party crate

<!--
Thank you for your submission!  You're helping make Diesel more robust 🎉

We'll try to respond as quickly as possible.
-->

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,12 @@ Increasing the minimal supported Rust version will always be coupled at least wi
 
 ## Unreleased 
 
+## [diesel_derives 2.0.2] 2023-03-13 
+
+## Fixed 
+
+* Fixing the fallout of a breaking change from `quote` by not using their internal API
+
 ## [2.0.3] 2023-01-24
 
 ## Fixed
@@ -1982,3 +1988,4 @@ queries or set `PIPES_AS_CONCAT` manually.
 [2.0.0]: https://github.com/diesel-rs/diesel/compare/v.1.4.0...v2.0.0
 [2.0.1]: https://github.com/diesel-rs/diesel/compare/v.2.0.0...v2.0.1
 [2.0.2]: https://github.com/diesel-rs/diesel/compare/v.2.0.1...v2.0.2
+[diesel_derives 2.0.2]: https://github.com/diesel-rs/diesel/compare/v.2.0.2...diesel_derives_v2.0.2

--- a/diesel_derives/Cargo.toml
+++ b/diesel_derives/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "diesel_derives"
-version = "2.0.1"
+version = "2.0.2"
 license = "MIT OR Apache-2.0"
 description = "You should not use this crate directly, it is internal to Diesel."
 documentation = "https://diesel.rs/guides/"

--- a/diesel_derives/src/attrs.rs
+++ b/diesel_derives/src/attrs.rs
@@ -2,11 +2,11 @@ use std::fmt::{Display, Formatter};
 
 use proc_macro2::{Span, TokenStream};
 use proc_macro_error::ResultExt;
-use quote::spanned::Spanned;
 use quote::ToTokens;
 use syn::parse::discouraged::Speculative;
 use syn::parse::{Parse, ParseStream, Parser, Result};
 use syn::punctuated::Punctuated;
+use syn::spanned::Spanned;
 use syn::token::Comma;
 use syn::{parenthesized, Attribute, Ident, LitBool, LitStr, Path, Type, TypePath};
 
@@ -141,7 +141,7 @@ impl Parse for FieldAttr {
 }
 
 impl Spanned for FieldAttr {
-    fn __span(&self) -> Span {
+    fn span(&self) -> Span {
         match self {
             FieldAttr::Embed(ident)
             | FieldAttr::ColumnName(ident, _)
@@ -240,7 +240,7 @@ impl Parse for StructAttr {
 }
 
 impl Spanned for StructAttr {
-    fn __span(&self) -> Span {
+    fn span(&self) -> Span {
         match self {
             StructAttr::Aggregate(ident)
             | StructAttr::NotSized(ident)
@@ -262,8 +262,6 @@ pub fn parse_attributes<T>(attrs: &[Attribute]) -> Vec<AttributeSpanWrapper<T>>
 where
     T: Parse + ParseDeprecated + Spanned,
 {
-    use syn::spanned::Spanned;
-
     attrs
         .iter()
         .flat_map(|attr| {

--- a/diesel_tests/tests/select.rs
+++ b/diesel_tests/tests/select.rs
@@ -1,4 +1,5 @@
 use super::schema::*;
+#[cfg(not(feature = "mysql"))]
 use crate::schema_dsl::*;
 use diesel::*;
 
@@ -214,7 +215,8 @@ table! {
     }
 }
 
-#[cfg(not(feature = "sqlite"))]
+// the test is somehow broken on some mariadb versions
+#[cfg(not(any(feature = "sqlite", feature = "mysql")))]
 #[test]
 fn select_for_update_locks_selected_rows() {
     use self::users_select_for_update::dsl::*;

